common/src/main/java/com/wynntils/models/seaskipper/SeaskipperModel.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.seaskipper;
+
+import com.google.common.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ContainerSetSlotEvent;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.mc.event.ScreenOpenedEvent;
+import com.wynntils.models.containers.ContainerModel;
+import com.wynntils.models.items.items.gui.SeaskipperDestinationItem;
+import com.wynntils.models.map.pois.SeaskipperPoi;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+public final class SeaskipperModel extends Model {
+    private static final StyledText OAK_BOAT_NAME = StyledText.fromString("§bOak Boat");
+    private static final Pattern SEASKIPPER_PASS_PATTERN = Pattern.compile("^§b(.*) Pass §7for §b(\\d+)²$");
+
+    private static AbstractContainerScreen actualSeaskipperScreen;
+
+    private final Map<SeaskipperDestinationItem, Integer> destinations = new HashMap<>();
```

Comment:
Ok, I'm beginning to understand this now. You are dealing with both a SeaskipperPoi and a SeaskipperDestinationItem for every destination. That makes things very confusing, since they both represent the same destination. They need to be brought together under one roof.

I think the best way is to create a new holder class (or even better a record), like `SeaskipperDestination` that has both a SeaskipperPoi and a SeaskipperDestinationItem as their field. Alternatively, the SeaskipperDestinationItem could be modified to hold on to a corresponding Poi, but I guess that makes for more changes throughout the code which will complicate things.